src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java

```diff
@@ -464,9 +465,38 @@ else if (groupIdx > lastGroup) {
      * @param line the line of the current import.
      * @return true if imports group are separated internally.
      */
-    private boolean checkSeparatorInGroup(int groupIdx, boolean isStatic, int line) {
-        return !beforeFirstImport && separated && groupIdx == lastGroup
-                && isStatic == lastImportStatic && line - lastImportLine > 1;
+    private boolean isSeparatorInGroup(int groupIdx, boolean isStatic, int line) {
+        final boolean inSameGroup = isInSameGroup(groupIdx, isStatic);
+        return (!separated || inSameGroup) && isThereSeparatorAfterLastImport(line);
+    }
+
+    /**
+     * Checks whether there is separator after last import.
+     * @param line the line of the current import.
+     * @return true if there is separator after last import.
+     */
+    private boolean isThereSeparatorAfterLastImport(int line) {
+        return !beforeFirstImport && line - lastImportLine > 1;
```

Comment:
isThereSeparatorAfterLastImport vs beforeFirstImport .
Method name and description is not reflecting real code - please name method properly